packages/babel-core/src/config/index.ts

```diff
@@ -24,62 +24,125 @@ export type {
 } from "./validation/options";
 
 import loadFullConfig, { type ResolvedConfig } from "./full";
-import { loadPartialConfig as loadPartialConfigRunner } from "./partial";
+import {
+  type PartialConfig,
+  loadPartialConfig as loadPartialConfigImpl,
+} from "./partial";
 
 export { loadFullConfig as default };
 export type { PartialConfig } from "./partial";
 
 import { createConfigItem as createConfigItemImpl } from "./item";
 import type { ConfigItem } from "./item";
 
-const loadOptionsRunner = gensync(function* (
-  opts: unknown,
-): Handler<ResolvedConfig | null> {
+import { beginHiddenCallStack } from "../errors/rewrite-stack-trace";
+
+const loadPartialConfigRunner = gensync(loadPartialConfigImpl);
+export function loadPartialConfigAsync(
+  ...args: Parameters<typeof loadPartialConfigRunner.async>
+) {
+  return beginHiddenCallStack(loadPartialConfigRunner.async)(...args);
+}
```

Comment:
Not for this PR, but I wonder if we could simplify our `beginHiddenCallStack` usage by doing something
```js
export const loadPartialConfigAsync = wrapHidingCallStack("loadPartialConfigAsync", loadPartialConfigRunner.async);
```

where
```ts
function wrapHidingCallStack<F extends (...args: unknown[]) => unknown>(name, fn: F): F {
  return {
    [name]: function () {
      return beginHiddenCallStack(fn).apply(this, arguments);
    },
  }[name];
}
```

Reply:
I think the current usage is fine, imo it is more straightforward than creating an object literal just for the sake of named evaluation.